lib/AppInfo/Application.php

```diff
@@ -218,4 +224,26 @@ protected function registerCollaborationResources(IProviderManager $resourceMana
 			Util::addScript('deck', 'deck-collections');
 		});
 	}
+
+	protected function registerSessionListener(IEventDispatcher $eventDispatcher, SessionService $sessionService): void {
+		$container = $this->getContainer();
+		try {
+			$queue = $container->get(IQueue::class);
+		} catch (\Exception $e) {
+			// most likely notify_push is not installed.
+			return;
+		}
+		
+		$eventDispatcher->addListener(SessionCreatedEvent::class, function (SessionCreatedEvent $event) use ($sessionService, $queue) {
```

Comment:
Maybe this could be moved to a dedicated Listener class that is registered using `$context->registerEventListener()` within the register method.